File: client/src/app/models/reservation.ts
export interface Reservation {
  id: number;
  reservableObjectId: number;
  objectName: string;
  objectType: string;
  locationId: number;
  locationName: string;
  username: string;
  startTime: string;
  endTime: string;
  status: 'Active' | 'Cancelled' | 'Completed';
  createdAt: string;
}

export interface CreateReservationRequest {
  reservableObjectId: number;
  startTime: string;
  endTime: string;
}

export interface ConflictCheck {
  hasConflict: boolean;
  message?: string;
  conflicts?: ConflictingReservation[];
}

export interface ConflictingReservation {
  id: number;
  startTime: string;
  endTime: string;
}

File: api/src/Workshop.Api/Data/Entities/Reservation.cs
namespace Workshop.Api.Data.Entities;

public class Reservation
{
    public int Id { get; set; }
    public int ReservableObjectId { get; set; }
    public string Username { get; set; } = string.Empty;
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public string Status { get; set; } = "Active";
    public DateTime CreatedAt { get; set; }
    
    public ReservableObject ReservableObject { get; set; } = null!;
}
